From: Trond Myklebust <trond.myklebust@hammerspace.com>
Date: Fri, 16 Aug 2019 08:58:48 -0400
Subject: [PATCH] SUNRPC: Handle connection breakages correctly in
 call_status()
Git-commit: c82e5472c9980e0e483f4b689044150eefaca408
Patch-mainline: v5.3
References: git-fixes

If the connection breaks while we're waiting for a reply from the
server, then we want to immediately try to reconnect.

Fixes: ec6017d90359 ("SUNRPC fix regression in umount of a secure mount")
Signed-off-by: Trond Myklebust <trond.myklebust@hammerspace.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 net/sunrpc/clnt.c |    1 +
 1 file changed, 1 insertion(+)

--- a/net/sunrpc/clnt.c
+++ b/net/sunrpc/clnt.c
@@ -2177,6 +2177,7 @@ call_status(struct rpc_task *task)
 	case -ECONNABORTED:
 	case -ENOTCONN:
 		rpc_force_rebind(clnt);
+		break;
 	case -EADDRINUSE:
 		rpc_delay(task, 3*HZ);
 	case -EPIPE:
